lib/slurm_squeue_client.rb

```diff
@@ -59,8 +67,8 @@ def squeue_jobs_running
   def sinfo
     return @sinfo if defined?(@sinfo)
 
-    cmd = '/usr/bin/sinfo'
-    args = ["-a", "-h", "-o=\"%C/%A/%D\"", "--clusters=\"#{@canonical_cluster_id}\""]
+    args = ["-a", "-h", "-o=\"%C/%A/%D\""]
+    args.push("--clusters=\"#{@canonical_cluster_id}\"") if defined?(@canonical_cluster_id)
```

Comment:
The `"--clusters=\"#{@canonical_cluster_id}\""` and `if defined?(@canonical_cluster_id)` is duplicated in multiple places. Might be more readable in one method:

```
def clusters_arg
  @clusters_arg ||= ( @canonical_cluster_id ? "--clusters=\"#{@canonical_cluster_id}\"" : ""
end
```

then

    args.push(clusters_arg) unless clusters_arg.blank?

and

    "#{sinfo_cmd} #{clusters_arg} -N -h -a ..."

Also, above addresses the issue that `@canonical_cluster_id` is always defined, so `if defined?(@canonical_cluster_id)` will always be true.